### PR TITLE
fix(NotificationListNew): Key and spacing issues

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

NotificationListNew touch-ups.

Fixes for monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/22305

1. Add horizontal padding on the dotted borders
2. Addresses the "each child does not have a unique key prop" error

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/KIqFTtIa7WoDA8G1owEvYL/%5BWIP%5D-Notifications?node-id=299%3A600
- [x] Related to JIRA ticket: [EGG-694](https://codecademy.atlassian.net/browse/EGG-694)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/NotificationListNew/index.tsx [hunk 1]
```diff
@@ -25,15 +29,14 @@ export const NotificationListNew: React.FC<NotificationListNewProps> = ({
       ) : (
         notifications.map((notification: Notification) => {
           return (
-            <>
+            <Box key={notification.id}>
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/NotificationListNew/index.tsx:32]: Do you specifically need an element? Would `<React.Fragment key={notification.id}>` work?

**Review by JoshuaKGoldberg — APPROVED** [review]

File: packages/gamut/src/NotificationListNew/index.tsx (hunk 1 above)
Comment (katiezutter) [reply packages/gamut/src/NotificationListNew/index.tsx:32]: Good call - yeah just needed to update the pattern padding I think!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@10.1.6-alpha.38a1c3.0<br/>@codecademy/gamut@25.2.2-alpha.38a1c3.0<br/>@codecademy/styleguide@29.3.4-alpha.38a1c3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://603513edab5e2820199a0b3c--gamut-preview.netlify.app>https://603513edab5e2820199a0b3c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/603513edab5e2820199a0b3c>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]

**Review by dreamwasp — APPROVED** [review]
🙏

**Pull request merged**